modules/lb-cloudscale/providers.tf

```diff
@@ -0,0 +1,22 @@
+terraform {
+  required_version = ">= 0.14"
+  experiments      = [module_variable_optional_attrs]
+  required_providers {
+    cloudscale = {
+      source  = "cloudscale-ch/cloudscale"
+      version = ">= 3.0"
+    }
+    null = {
+      source  = "hashicorp/null"
+      version = ">= 2.1"
+    }
```

Comment:
I don't think you need provider null anymore for the lb module, since those resources are in the vshn hieradata module.